chain/chain/src/chain.rs

```diff
@@ -1287,6 +1322,9 @@ impl Chain {
             if header.challenges_root() != &MerkleHash::default() {
                 return Err(Error::InvalidChallengeRoot);
             }
+
+            // Check if block can be finalized.
+            self.check_if_finalizable(&header)?;
```

Comment:
I think it makes more sense to move this function to be called in "Chain::preprocess_block" because 
1) Logically, the `validate_header` function checks if a header is valid or invalid, whether the block is on an old fork is not relevant here in this function. You can see all the other checks that this functions is about the validity of the header. 
2) It is better to have the check the closest to when the block is about to be processed. Technically, this check is just a pre-check before the `apply_chunks` function calls `get_ref`. Even if we check that the block is not on an old fork now, the `get_ref` function may still fail later because the final head can be moved between when `check_if_finalizable` and `get_ref` are called. So we want to shorten the time in between to reduce the probability that happens. Note that we still need to handle the error from `get_ref` correctly even if `check_if_finalizable` is called.